src/org/traccar/protocol/MiniFinderProtocolDecoder.java

```diff
@@ -34,44 +34,145 @@ public MiniFinderProtocolDecoder(MiniFinderProtocol protocol) {
+            .expression("!A,")
             .number("(d+)/(d+)/(d+),")           // date
             .number("(d+):(d+):(d+),")           // time
             .number("(-?d+.d+),")                // latitude
             .number("(-?d+.d+),")                // longitude
-            .number("(d+.?d*),")                 // speed
+            .number("(d+.?d*),")                 // unknown
+            .number("(d+.?d*),")                 // unknown
+            .number("(d+.?d*)")                  // unknown
+            .compile();
+
+    // The !C record is not in the documentation for MiniFinder or EV-07S
+    // Eview, the manufacturer, indicated that resellers could implement new record types.
+    // Using the data in MiniFinderProtocolDecoderTest.java as a guide
+    private static final Pattern PATTERN_C = new PatternBuilder()
+            .expression("!C,")
+            .number("(d+)/(d+)/(d+),")           // date
+            .number("(d+):(d+):(d+),")           // time
+            .number("(-?d+.d+),")                // latitude
+            .number("(-?d+.d+),")                // longitude
+            .number("(d+.?d*),")                 // speed (km/h)
+            .number("(d+.?d*),")                 // course
+            .number("(x+),")                     // flags
+            .number("(-?d+.d+),")                // altitude (meters)
+            .number("(d+),")                     // battery (percentage)
+            .number("(d+.?d*),")                 // unknown
+            .number("(d+.?d*),")                 // unknown
+            .number("(d+.?d*)")                  // unknown
+            .compile();
+
+    // The !B (buffered data) records are the same as !D (live data) records.
+    // This was confirmed with the manufacturer, Eview.
+    private static final Pattern PATTERN_BD = new PatternBuilder()
+            .expression("![BD],")
+            .number("(d+)/(d+)/(d+),")           // date
+            .number("(d+):(d+):(d+),")           // time
+            .number("(-?d+.d+),")                // latitude
+            .number("(-?d+.d+),")                // longitude
+            .number("(d+.?d*),")                 // speed (km/h)
             .number("(d+.?d*),")                 // course
-            .groupBegin()
             .number("(x+),")                     // flags
-            .number("(-?d+.d+),")                // altitude
-            .number("(d+),")                     // battery
+            .number("(-?d+.d+),")                // altitude (meters)
+            .number("(d+),")                     // battery (percentage)
             .number("(d+),")                     // satellites in use
             .number("(d+),")                     // satellites in view
-            .text("0")
-            .or()
-            .any()
-            .groupEnd()
+            .number("(d+.?d*)")                  // HDOP, confirmed with Eview manufacturer
```

Comment:
Comments are usually lower case. "Confirmed with Eview manufacturer" is unnecessary. You can mention it in the pull request comment for future reference.